amoro-ams/src/main/java/org/apache/amoro/server/dashboard/controller/OptimizerController.java

```diff
@@ -75,10 +76,15 @@ public void getOptimizerTables(Context ctx) {
     List<TableRuntime> tableRuntimes =
         tableRuntimeBeans.stream()
             .map(meta -> tableService.getRuntime(meta.getTableId()))
+            .filter(Objects::nonNull)
```

Comment:
Since we are using the in-memory table runtime list, why do we still need to query the ID list from the database and get list count from memory? 
It seems quite fragmented

Reply:
Yes, this is defensive programming, because we will process the information we get here(TableRuntime list) later and throw Exception in `OptimizingUtil#buildTableOptimizeInfo`), of course, we can also remove it here if needed